lib/rules/no-useless-escape.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @fileoverview Look for useless escapes in strings and regexes
+ * @author Onur Temizkan
+ * @copyright 2016 Onur Temizkan. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var VALID_STRING_ESCAPES = [
+    "\\",
+    "n",
+    "r",
+    "v",
+    "t",
+    "b",
+    "f",
+    "u",
+    "x"
+];
+
+var VALID_REGEX_ESCAPES = [
+    "\\",
+    ".",
+    "-",
+    "^",
+    "$",
+    "*",
+    "+",
+    "?",
+    "{",
+    "}",
+    "[",
+    "]",
+    "|",
+    "(",
+    ")",
+    "b",
+    "c",
+    "d",
+    "D",
+    "f",
+    "n",
+    "r",
+    "s",
+    "S",
+    "t",
+    "v",
+    "w",
+    "W",
+    "x",
+    "u"
+];
+
+module.exports = function(context) {
+    /**
+     * Checks if the escape character in given slice is unnecessary.
+     *
+     * @private
+     * @param {string} elm - string slice to validate.
+     * @param {ASTNode} node - node to validate.
+     * @returns {void}
+     * @this escapes_quote_node
+     */
+    function validate(elm) {
+        if ((this.escapes.indexOf(elm[1]) === -1) && !(this.quote && (elm[1] === this.quote))) {
```

Comment:
A bunch of these parens are unnecessary.

```js
if (this.escapes.indexOf(elm[1]) === -1 && !(this.quote && elm[1] === this.quote)) {
```

Also, you might just want to pull out some of those expressions as variables and name them.